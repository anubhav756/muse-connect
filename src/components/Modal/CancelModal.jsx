import React, { Component } from 'react';
import Dialog from 'material-ui/Dialog';
import colors from '!!sass-variable-loader!./../../styles/variables/colors.scss';
import RaisedButton from 'material-ui/RaisedButton';

let _alert = () => Promise.resolve();
const alert = warningText => _alert(warningText);
export default alert;

export class CancelModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      warningText: null,
      promise: null
    }
    this.handleClose = this.handleClose.bind(this);
    _alert = this.trigger.bind(this);
  }
  trigger(warningText) {
    return new Promise((resolve, reject) => {
      this.setState({
        promise: {
          resolve,
          reject
        },
        warningText,
        open: true
      })
    });
  }
  handleClose() {
    this.setState({
      open: false
    })
  }
  render() {
    const actions = [
      <RaisedButton
        label="Yes"
        primary
        style={{ marginRight: '10px', borderRadius: '10px' }}
        buttonStyle={{ borderRadius: '10px' }}
        overlayStyle={{ borderRadius: '10px' }}
        onClick={() => { this.handleClose(); this.state.promise.resolve() }}
      />,
      <RaisedButton
        label="No"
        primary
        keyboardFocused
        style={{ borderRadius: '10px' }}
        buttonStyle={{ borderRadius: '10px' }}
        overlayStyle={{ borderRadius: '10px' }}
        onClick={() => { this.handleClose(); this.state.promise.reject() }}
      />
    ];

    return (
      <Dialog
        actions={actions}
        modal={false}
        open={this.state.open}
        onRequestClose={this.handleClose}
        style={{ zIndex: 10001 }}
        overlayStyle={{ backgroundColor: 'rgba(255,255,255,0.8)' }}
      >
        <div style={{ fontFamily: 'proxima_novamedium', color: colors.mediumGrey, }}>
          {this.state.warningText}
        </div>
      </Dialog>
    );
  }
}
