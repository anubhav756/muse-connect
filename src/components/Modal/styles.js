import colors from '!!sass-variable-loader!./../../styles/variables/colors.scss';

module.exports = {
  modal: {
    zIndex: 10001
  },
  title: {
    fontFamily: 'proxima_novabold',
    color: colors.darkGrey,
    fontSize: '20px',
    paddingLeft: '42px',
    paddingRight: '48px'
  },
  subTitle: {
    fontFamily: 'proxima_novaregular',
    color: colors.darkGrey,
    fontSize: '18px'
  },
  titleContainer: {
    marginRight: '0px'
  },
  xIcon: {
    position: 'absolute',
    top: 12,
    right: 12
  },
  iconContainer: {
    width: 16,
    height: 16
  },
  bodyStyle: {
    paddingLeft: 42,
    paddingRight: 26,
    marginRight: 16,
  },
  overlayStyle: {
    backgroundColor: 'rgba(255,255,255,0.8)'
  }
}
