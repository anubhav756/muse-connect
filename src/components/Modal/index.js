import Modal from './Modal';
import alert, { CancelModal } from './CancelModal';

export { alert, CancelModal };
export default Modal;
