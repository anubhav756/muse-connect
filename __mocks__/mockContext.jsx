import React from 'react';
import PropTypes from 'prop-types'
import { getMuiTheme } from 'material-ui/styles'

export function mockStore(dummyStore = {}, dispatch = {}) {
  return {
    getState: () => dummyStore,
    dispatch: () => dispatch,
    subscribe: () => {}
  }
}

export default function (dummyStore, dispatch) {
  return {
    context: {
      store: mockStore(dummyStore, dispatch),
      muiTheme: getMuiTheme(),
    },
    childContextTypes: {
      muiTheme: PropTypes.object,
      store: PropTypes.object
    }
  }
}
