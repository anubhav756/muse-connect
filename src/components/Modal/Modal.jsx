import React from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import IconButton from 'material-ui/IconButton';
import styles from './styles';
import Icon from '../Icon';

function Modal({
  title,
  subTitle,
  open,
  toggleModal,
  children,
  iconName,
  iconStyle,
  titleStyle,
  subTitleStyle,
  autoScrollBodyContent,
  style,
  actions,
  containerStyle,
  contentClassName,
  childrenStyle,
  childrenClassName,
  showCross,
  ...otherprops
}) {
  return (
    <Dialog
      title={
        <div style={styles.titleContainer}>
          {title}
          {subTitle && <div style={subTitleStyle}>{subTitle}</div>}
          {
            showCross &&
            <IconButton
              iconStyle={{ height: 16, width: 16 }}
              style={{ ...styles.xIcon, ...iconStyle }}
              onClick={toggleModal}
            >
              <Icon name={iconName} style={styles.iconContainer} />
            </IconButton>
          }
        </div>
      }
      titleStyle={{ ...styles.title, ...titleStyle }}
      contentClassName={contentClassName}
      open={open}
      onRequestClose={toggleModal}
      autoScrollBodyContent={autoScrollBodyContent}
      style={{ ...styles.modal, ...containerStyle }}
      contentStyle={style}
      bodyStyle={{ ...styles.bodyStyle, ...childrenStyle }}
      bodyClassName={`scrollbarModal ${childrenClassName}`}
      overlayStyle={styles.overlayStyle}
      actions={actions}
      {...otherprops}
    >
      {children}
    </Dialog>
  );
}

Modal.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  open: PropTypes.bool.isRequired,
  toggleModal: PropTypes.func.isRequired,
  iconName: PropTypes.string,
  contentClassName: PropTypes.string,
  titleStyle: PropTypes.object,
  subTitleStyle: PropTypes.object,
  autoScrollBodyContent: PropTypes.bool,
  actions: PropTypes.array,
  style: PropTypes.object,
  iconStyle: PropTypes.object,
  containerStyle: PropTypes.object,
  childrenStyle: PropTypes.object,
  childrenClassName: PropTypes.string,
  showCross: PropTypes.bool
}
Modal.defaultProps = {
  subTitle: null,
  subTitleStyle: styles.subTitle,
  iconName: 'x-icon',
  titleStyle: {},
  autoScrollBodyContent: true,
  style: null,
  actions: null,
  containerStyle: {},
  childrenStyle: {},
  contentClassName: '',
  showCross: true,
  iconStyle: {},
  childrenClassName: ''
}

export default Modal;
