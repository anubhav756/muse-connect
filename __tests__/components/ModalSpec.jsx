import React from 'react';
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json';
import Modal, { CancelModal, alert } from '../../src/components/Modal';
import mockContext from '../../__mocks__/mockContext';


describe('<Modal />', () => {
  const buildWrapper = (props, store, isCancelModal) => (
    mount(isCancelModal ? <CancelModal {...props} /> : <Modal {...props} />, mockContext(store))
  )

  it('<Modal /> component', () => {
    const component = buildWrapper({
      title: 'test modal',
      open: true,
      toggleModal: () => {}
    });
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('<CancelModal /> renders correctly', () => {
    const component = buildWrapper({
      title: 'test modal',
      open: true,
      toggleModal: () => {}
    },
    null,
    true
    );
    expect(toJSON(component)).toMatchSnapshot();
    alert('This is a test');
  });

  it('<CancelModal /> handles yes', () => {
    const component = buildWrapper(null, null, true);
    alert('This is a test');
    component.find('Dialog').getElement().props.actions[0].props.onClick();
    expect(toJSON(component)).toMatchSnapshot();
  });
  it('<CancelModal /> handles no', () => {
    const component = buildWrapper(null, null, true);
    alert('This is a test');
    expect(toJSON(component)).toMatchSnapshot();
    component.find('Dialog').getElement().props.actions[1].props.onClick();
  });
});
