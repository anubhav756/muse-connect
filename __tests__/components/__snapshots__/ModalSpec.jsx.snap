// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> <CancelModal /> handles no 1`] = `
<CancelModal>
  <Dialog
    actions={
      Array [
        <RaisedButton
          buttonStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          disabled={false}
          fullWidth={false}
          label="Yes"
          labelPosition="after"
          onClick={[Function]}
          overlayStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          primary={true}
          secondary={false}
          style={
              Object {
                  "borderRadius": "10px",
                  "marginRight": "10px",
                }
          }
      />,
        <RaisedButton
          buttonStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          disabled={false}
          fullWidth={false}
          keyboardFocused={true}
          label="No"
          labelPosition="after"
          onClick={[Function]}
          overlayStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          primary={true}
          secondary={false}
          style={
              Object {
                  "borderRadius": "10px",
                }
          }
      />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    overlayStyle={
      Object {
        "backgroundColor": "rgba(255,255,255,0.8)",
      }
    }
    repositionOnUpdate={true}
    style={
      Object {
        "zIndex": 10001,
      }
    }
  >
    <RenderToLayer
      open={true}
      render={[Function]}
      useLayerForClickAway={false}
    />
  </Dialog>
</CancelModal>
`;

exports[`<Modal /> <CancelModal /> handles yes 1`] = `
<CancelModal>
  <Dialog
    actions={
      Array [
        <RaisedButton
          buttonStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          disabled={false}
          fullWidth={false}
          label="Yes"
          labelPosition="after"
          onClick={[Function]}
          overlayStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          primary={true}
          secondary={false}
          style={
              Object {
                  "borderRadius": "10px",
                  "marginRight": "10px",
                }
          }
      />,
        <RaisedButton
          buttonStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          disabled={false}
          fullWidth={false}
          keyboardFocused={true}
          label="No"
          labelPosition="after"
          onClick={[Function]}
          overlayStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          primary={true}
          secondary={false}
          style={
              Object {
                  "borderRadius": "10px",
                }
          }
      />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    overlayStyle={
      Object {
        "backgroundColor": "rgba(255,255,255,0.8)",
      }
    }
    repositionOnUpdate={true}
    style={
      Object {
        "zIndex": 10001,
      }
    }
  >
    <RenderToLayer
      open={true}
      render={[Function]}
      useLayerForClickAway={false}
    />
  </Dialog>
</CancelModal>
`;

exports[`<Modal /> <CancelModal /> renders correctly 1`] = `
<CancelModal
  open={true}
  title="test modal"
  toggleModal={[Function]}
>
  <Dialog
    actions={
      Array [
        <RaisedButton
          buttonStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          disabled={false}
          fullWidth={false}
          label="Yes"
          labelPosition="after"
          onClick={[Function]}
          overlayStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          primary={true}
          secondary={false}
          style={
              Object {
                  "borderRadius": "10px",
                  "marginRight": "10px",
                }
          }
      />,
        <RaisedButton
          buttonStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          disabled={false}
          fullWidth={false}
          keyboardFocused={true}
          label="No"
          labelPosition="after"
          onClick={[Function]}
          overlayStyle={
              Object {
                  "borderRadius": "10px",
                }
          }
          primary={true}
          secondary={false}
          style={
              Object {
                  "borderRadius": "10px",
                }
          }
      />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    overlayStyle={
      Object {
        "backgroundColor": "rgba(255,255,255,0.8)",
      }
    }
    repositionOnUpdate={true}
    style={
      Object {
        "zIndex": 10001,
      }
    }
  >
    <RenderToLayer
      open={true}
      render={[Function]}
      useLayerForClickAway={false}
    />
  </Dialog>
</CancelModal>
`;

exports[`<Modal /> <Modal /> component 1`] = `
<Modal
  actions={null}
  autoScrollBodyContent={true}
  childrenClassName=""
  childrenStyle={Object {}}
  containerStyle={Object {}}
  contentClassName=""
  iconName="x-icon"
  iconStyle={Object {}}
  open={true}
  showCross={true}
  style={null}
  subTitle={null}
  subTitleStyle={
    Object {
      "color": undefined,
      "fontFamily": "proxima_novaregular",
      "fontSize": "18px",
    }
  }
  title="test modal"
  titleStyle={Object {}}
  toggleModal={[Function]}
>
  <Dialog
    actions={null}
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    bodyClassName="scrollbarModal "
    bodyStyle={
      Object {
        "marginRight": 16,
        "paddingLeft": 42,
        "paddingRight": 26,
      }
    }
    contentClassName=""
    contentStyle={null}
    modal={false}
    onRequestClose={[Function]}
    open={true}
    overlayStyle={
      Object {
        "backgroundColor": "rgba(255,255,255,0.8)",
      }
    }
    repositionOnUpdate={true}
    style={
      Object {
        "zIndex": 10001,
      }
    }
    title={
      <div
        style={
          Object {
            "marginRight": "0px",
          }
        }
      >
        test modal
        <IconButton
          disableTouchRipple={false}
          disabled={false}
          iconStyle={
            Object {
              "height": 16,
              "width": 16,
            }
          }
          onClick={[Function]}
          style={
            Object {
              "position": "absolute",
              "right": 12,
              "top": 12,
            }
          }
          tooltipPosition="bottom-center"
          touch={false}
        >
          <Icon
            name="x-icon"
            style={
              Object {
                "height": 16,
                "width": 16,
              }
            }
          />
        </IconButton>
      </div>
    }
    titleStyle={
      Object {
        "color": undefined,
        "fontFamily": "proxima_novabold",
        "fontSize": "20px",
        "paddingLeft": "42px",
        "paddingRight": "48px",
      }
    }
  >
    <RenderToLayer
      open={true}
      render={[Function]}
      useLayerForClickAway={false}
    />
  </Dialog>
</Modal>
`;
